packages/@expo/cli/src/start/server/UrlCreator.ts

```diff
@@ -158,7 +157,7 @@ function joinUrlComponents({ protocol, hostname, port }: Partial<UrlComponents>)
   // This is because Android React Native WebSocket implementation is not spec compliant and fails without a port:
   // `E unknown:ReactNative: java.lang.IllegalArgumentException: Invalid URL port: "-1"`
   // Invoked first in `metro-runtime/src/modules/HMRClient.js`
-  const validPort = env.EXPO_NO_DEFAULT_PORT ? port : port || '80';
+  const validPort = port ?? '80';
```

Comment:
if the port will be non-nullable string, we may rewrite the code as:
```js
  const validProtocol = protocol ? `${protocol}://` : '';

  let url = `${validProtocol}${hostname}`;
  if (port) {
    url += `:${port}`;
  }
  return url;
```

at first glance, i thought the port is undefined and the `const validPort = port ?? '80';` will always add the port 80. after testing, i just figured out the port is '' in the case.

we could also remove the comment
```
  // Android HMR breaks without this port 80.
  // This is because Android React Native WebSocket implementation is not spec compliant and fails without a port:
  // `E unknown:ReactNative: java.lang.IllegalArgumentException: Invalid URL port: "-1"`
  // Invoked first in `metro-runtime/src/modules/HMRClient.js`
```